src/MOI_wrapper.jl

```diff
@@ -0,0 +1,414 @@
+function MOI.set(o::Optimizer, ::MOI.VariableName, vi::VI, name::String)
+    @SC SCIPchgVarName(get_scip(o), get_var(o, vi), name)
+end
+
+scip_vartype(::Type{BINS}) = SCIP_VARTYPE_BINARY
+scip_vartype(::Type{INTS}) = SCIP_VARTYPE_INTEGER
+function MOI.add_constraint(o::Optimizer, func::SVF, set::S) where {S <: TYPES}
+    allow_modification(o)
+    scip = get_scip(o)
+    var = get_var(o, func.variable)
+    infeasible = Ref{Ptr{SCIP_Bool}}
+    @SC SCIPchgVarType(scip, var, scip_vartype(S), infeasible[])
+    if S <: BINS
+        # need to adjust bounds for SCIP?!
+        @SC SCIPchgVarLb(scip, var, 0.0)
+        @SC SCIPchgVarUb(scip, var, 1.0)
+    end
+    # use var index for cons index of this type
+    i = func.variable.value
+    return register!(o, CI{SVF, S}(i))
+end
+
+function MOI.add_constraint(o::Optimizer, func::SVF, set::S) where S <: BOUNDS
+    allow_modification(o)
+    var = get_var(o, func.variable)
+    lb, ub = bounds(set)
+    lb == nothing || @SC SCIPchgVarLb(get_scip(o), var, lb)
+    ub == nothing || @SC SCIPchgVarUb(get_scip(o), var, ub)
+    # use var index for cons index of this type
+    i = func.variable.value
+    return register!(o, CI{SVF, S}(i))
+end
+
+function MOI.set(o::SCIP.Optimizer, ::MOI.ConstraintSet, ci::CI{SVF,S}, set::S) where {S <: BOUNDS}
+    allow_modification(o)
+    var = get_var(o, VI(ci.value)) # cons index is actually var index
+    lb, ub = bounds(set)
+    lb == nothing || @SC SCIPchgVarLb(get_scip(o), var, lb)
+    ub == nothing || @SC SCIPchgVarUb(get_scip(o), var, ub)
+    return nothing
+end
+
+function MOI.is_valid(o::Optimizer, ci::CI{SVF,<:BOUNDS})
+    1 <= ci.value <= length(o.mscip.vars)
+end
+
+function MOI.add_constraint(o::Optimizer, func::SAF, set::S) where {S <: BOUNDS}
+    if func.constant != 0.0
+        msg = "SCIP does not support linear constraints with a constant offset."
+        throw(MOI.AddConstraintNotAllowed{SAF, S}(msg))
+    end
+
+    allow_modification(o)
+    scip = get_scip(o)
+
+    varidx = [t.variable_index.value for t in func.terms]
+    coefs = [t.coefficient for t in func.terms]
+
+    lhs, rhs = bounds(set)
+    lhs = lhs == nothing ? -SCIPinfinity(scip) : lhs
+    rhs = rhs == nothing ?  SCIPinfinity(scip) : rhs
+
+    i = add_linear_constraint(o.mscip, varidx, coefs, lhs, rhs)
+    ci = CI{SAF, S}(i)
+    register!(o, ci)
+    register!(o, get_cons(o, ci), i)
+    return ci
+end
+
+function MOI.set(o::SCIP.Optimizer, ::MOI.ConstraintSet, ci::CI{SAF,S}, set::S) where {S <: BOUNDS}
+    allow_modification(o)
+    scip = get_scip(o)
+    cons = get_cons(o, ci)
+
+    lhs, rhs = bounds(set)
+    lhs = lhs == nothing ? -SCIPinfinity(scip) : lhs
+    rhs = rhs == nothing ?  SCIPinfinity(scip) : rhs
+
+    @SC SCIPchgLhsLinear(scip, cons, lhs)
+    @SC SCIPchgRhsLinear(scip, cons, rhs)
+
+    return nothing
+end
+
+function MOI.is_valid(o::Optimizer, ci::CI{SAF,<:BOUNDS})
+    1 <= ci.value <= length(o.mscip.cons)
```

Comment:
Nit: Use explicit `return` except for one-line functions (https://github.com/JuliaOpt/JuMP.jl/issues/1718).